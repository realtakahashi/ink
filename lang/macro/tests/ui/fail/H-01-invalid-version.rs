use ink_lang as ink;

#[ink::contract(version = "foo")]
mod invalid_version {
    #[ink(storage)]
    struct InvalidVersion {}

    impl InvalidVersion {
        #[ink(constructor)]
        fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        fn message(&self) {}
    }
}

fn main() {}
