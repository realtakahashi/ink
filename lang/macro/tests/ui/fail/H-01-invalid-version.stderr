error: couldn't match provided version as semantic version string: foo.
info: The format expects a number triplet x.y.z, e.g. "0.1.0"
 --> $DIR/H-01-invalid-version.rs:3:27
  |
3 | #[ink::contract(version = "foo")]
  |                           ^^^^^
